src/extension/index.ts

```diff
@@ -1,15 +1,41 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-import { ExtensionContext } from 'vscode';
-import * as path from 'path';
+import { Event, EventEmitter, ExtensionContext, notebooks, extensions, NotebookEditor } from 'vscode';
+import { OpenImageInPlotViewer, SaveImageAs, IsJupyterExtensionInstalled } from './constants';
+
+export async function activate(
+    context: ExtensionContext
+): Promise<{
+    onDidReceiveMessage: Event<{ editor: NotebookEditor; message: OpenImageInPlotViewer | SaveImageAs }>;
+}> {
+    const onDidReceiveMessage = new EventEmitter<{
+        editor: NotebookEditor;
+        message: OpenImageInPlotViewer | SaveImageAs;
+    }>();
+    const messaging = notebooks.createRendererMessaging('jupyter-notebook-renderer');
+    context.subscriptions.push(
+        messaging.onDidReceiveMessage(({ editor, message }) => {
+            const msg = message as OpenImageInPlotViewer | SaveImageAs | IsJupyterExtensionInstalled;
+            if (!msg.type) {
+                return;
+            }
+            if (msg.type === 'isJupyterExtensionInstalled') {
+                const isJupyterExtAvailable = extensions.getExtension('ms-toolsai.jupyter');
+                void messaging.postMessage(
+                    <IsJupyterExtensionInstalled>{
+                        type: 'isJupyterExtensionInstalled',
+                        response: isJupyterExtAvailable
+                    },
+                    editor
+                );
+                return;
+            }
+            onDidReceiveMessage.fire({ editor, message: msg });
```

Comment:
You might also post a command? Then jupyter doesn't need to listen to this event. It can just register the command handler? That would mean nobody else could register the command listener though

Reply:
> That would mean nobody else could register the command listener though

Not really, if extension `A` registers, first, then we won't be able to register such a command.
Only one extension can register a command.

Reply:
Yeah that's what I meant. If somebody else registers the command before jupyter, then it won't work. But I think that's unlikely. 

Command is looser coupling (both good and bad I'd say)

Reply:
I'd like to leave this, let's discuss cleaning later, I have an issue for that (we can bring them into this extension, saving and plot viewer)